### ISSUE TITLE
ERROR Infinite radiance value returned for pixel  in 'infinite' light type.

### ISSUE THREAD

**Issue by pbrt4bounty** [mmp/pbrt-v4#130]
When try to defined a 'Infinite' light type with 'illuminance' and 'scale' options, I get a lot of error messages like this.....
```
[ tid 003 @ 7.990s ..\src\pbrt\cpu\integrators.cpp:271 ] 
    ERROR Infinite radiance value returned for pixel (557, 106), sample 0. Setting to black. 
```
There are the definition..
```
AttributeBegin
    LightSource "distant"
        "point3 from" [0.0 -3.63 4.895448]
	"point3 to" [-0.544639 -3.63 4.056777]
	"float illuminance" [ 2.0 ]
	"rgb L" [ 1.0 1.0 1.0 ]
	"float scale" [ 0.1 ]
AttributeEnd
```

Am I defining this type of light properly? With only 'scale' option, without 'illuminance' the render work fine. 
Cheers..

**Comment by mmp** [comment]
Interesting--thanks for reporting this!

The culprit is the first line of [this if statement](https://github.com/mmp/pbrt-v4/blob/e2c6959fcaff5e7db0d1be1003812c9253939e00/src/pbrt/lights.cpp#L262):
```
    // Adjust scale to meet target illuminance value
    // Like for IBLs we measure illuminance as incident on an upward-facing
    // patch.    
    Float E_v = parameters.GetOneFloat("illuminance", -1);
    if (E_v > 0) {
        Float k_e = -w.y;
        sc *= E_v / k_e;
    }
```
Your light direction happens to have `w.y == 0`, and then things go bad from there. One issue is that the code should be using `w.z`, though the bug would still be present for a light with `w.z == 0` in that case.

I'm wondering if we should just remove the divide by `k_e` entirely, since it seems potentially problematic in multiple ways (also including a downward-facing direction.) What do you think, @anderslanglands?

**Comment by povmaniaco** [comment]
Indeed, removing the division by 'k_e' the error message disappears..
Cheers..
